src/util/output.c

```diff
@@ -14,9 +14,9 @@ static HANDLE output_filehandle = INVALID_HANDLE_VALUE;
 void print(char* str, ...)
 {
     va_list args;
-    char buffer[500];
+    char buffer[512];
```

Comment:
Could a comment be added explaining why the buffer size needs to be 512? (Mentioning the circumstances when this size is relevant helps future maintenance if and when those circumstances change)

Reply:
length of 512 == 0x200
length of 500 == 0x1F4

Not sure why it's at 500 length, it's best to follow what we are really used to. Although, below may could limit to 510 in order to append "\n" to the string. Which leave the last part as null terminated.

Reply:
Okay, well if it was known exactly which code calls this function and with which inputs, a maximum required length could be derived from that. But if that's infeasible or otherwise deemed not worth the time, then just keep it like this.

However, if this gets changed, it might be nice to declare the length in a const, and use that for both the application size and (reduced by 2) for the limit argument to the `vsnprintf` call

Reply:
> Okay, well if it was known exactly which code calls this function and with which inputs, a maximum required length could be derived from that. But if that's infeasible or otherwise deemed not worth the time, then just keep it like this.

Not possible, these are prefix format string with dynamic inputs. Yet widen the buffer length dynamically is too much effort to constant double checking for standard library's __printf call if we had allocated enough or not.

> However, if this gets changed, it might be nice to declare the length in a const, and use that for both the buffer allocation size and (reduced by 2) for the limit argument to the `vsnprintf` call

This is the only place we format the string from the buffer cache. I don't see the need for declare the length in a const? Technically we "could" lower it down to 256 length. Also, nxdk's debugPrint is internally limited to 512 length currently.

Reply:
> Not possible, these are prefix format string with dynamic inputs. Yet widen the buffer length dynamically is too much effort to constant double checking for standard library's __printf call if we had allocated enough or not.

This is only being called by this project, so it should be possible to determine all possible inputs.
But indeed, a dynamic buffer would be impractical.

> This is the only place we format the string from the buffer cache. I don't see the need for declare the length in a const? Technically we "could" lower it down to 256 length. Also, nxdk's debugPrint is internally limited to 512 length currently.

A const would ease maintenance, that's all.

I'm not going to press this line if thought further, as you seem set on keeping it like this.

And since I'm fine either way, I'll just approve it.

Reply:
Are you referring to like enum type of variable to be used for initial variable and limited range to format the string?

If so, then very well. I will update it.